### DISCUSSION TITLE
How to draw the h3map2004 map(colored)?

### DISCUSSION THREAD

**Discussion by ZhaoZheLin** [acorg/Racmacs#167] [Q&A]
Hello all,   
    I saw that the original h3map2004 map **was colored by clusters,** but when I drew the map, all the points were green, plus the map was different every time I ran it.
    In addition, I saw your reply to Shaoyuan that the agFill() and srFill() functions can only assign values to the color of a single point. Do you know a way to change colors by clusters like in the map ? I want to redraw the h3map2004 map.
    Thank you for taking the time to answer my questions!

**Comment by drserajames** [comment]
There are two parts to this question
1. How to colour the 2004 map?
2. Why is it different when rerun?

The first part is simpler - the 2004 is included in the package as an ace file. So you can either optimise that map or apply the plot specifications from that map to the one who have run.

```
# get data 
map <- read.acmap(paste0(system.file("extdata", package = "Racmacs"), "/h3map2004.ace"))
view(map)

# making a map (only 10 optimisation) & colouring as 2004 map
map2 <- optimizeMap(map, number_of_dimensions = 2, number_of_optimizations = 10)
view(map2)
map3 <- make.acmap(titerTable(map), number_of_dimensions = 2, number_of_optimizations = 10)
map3 <- applyPlotspec(map3, map)
view(map3)
```

The second part is a bit more complicated. The mds algorithm uses random starting coordinates for the points, different ones for each run. So each run may converge on a different local optimum. Depending on your data and optimisation parameter, you may find that the the map is very similar for most of the optimisation runs. But sometimes, as with the 2004 map, there are differences. Also sometimes there are small differences in the stress, but the maps themselves look very similar. You can use `set.seed()` before running an optimisation if you want to make sure you always get the same results. Is that sufficient to answer your question?

**Comment by judyssister** [comment]
Hi Sarah. Thanks for kind help. Could please let me know that what's the difference between optimizeMap() and make.acmap()?
